service/collector.go

```diff
@@ -99,14 +100,26 @@ func New(set CollectorSettings) (*Collector, error) {
 	}
 
 	col := &Collector{
-		info:         set.BuildInfo,
-		factories:    set.Factories,
-		stateChannel: make(chan State, Closed+1),
+		info:           set.BuildInfo,
+		factories:      set.Factories,
+		stateChannel:   make(chan State, Closed+1),
+		parserProvider: set.ParserProvider,
+		configLoader:   set.ConfigLoader,
 		// We use a negative in the settings not to break the existing
 		// behavior. Internally, allowGracefulShutodwn is more readable.
 		allowGracefulShutodwn: !set.DisableGracefulShutdown,
 	}
 
+	if col.parserProvider == nil {
+		// use default provider.
+		col.parserProvider = parserprovider.Default()
+	}
+
+	if col.configLoader == nil {
+		// use default provider.
+		col.configLoader = configloader.NewDefaultLoader()
+	}
```

Comment:
Can we document the relationship between ConfigLoader and ParserProvider? I think it is becoming a bit confusing. A documentation explaining how config works in general would help a lot.
Can be a future PR.

Reply:
I will do in a future PR, since ConfigSource may also be confusing. I will try to explain that better.